### PR TITLE
Add components section and first Action list component doc

### PR DESCRIPTION
**Pull request by yaili**
To-do:
- [x] Add a components section and links to new pages in the nav
- [x] Add pages for at least 1 overlay component
- [x] Fake metadata so it cross-links to implementation docs (for overlay components this should be React and Figma)
- [x] ~~Do we need to create new doctocat components for the component templates to work?~~
- [x] Fix empty pages (UI patterns, foundations, design tools + components)
- [ ] ~~Maybe: Quick update to landing page to introduce the different sections and content~~

### CODE REVIEW

File: content/components/action-list.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+<!--
+<ButtonOutline mr={1}>React component</ButtonOutline>
+<ButtonOutline mr={1}>ViewComponent</ButtonOutline>
+<ButtonOutline>Figma component</ButtonOutline>
+-->
+
+## Overview
+
+Action lists can have many applications. They’re the foundation of menus, selection panels, and other overlay components; they can be applied to page sidebars for showing individual actions, for handling local navigation, and for displaying metadata.
```
Comment (yaili) [inline content/components/action-list.mdx:22]:
```suggestion
Action lists can have many applications:
- They’re the foundation of menus, selection panels, and other overlay components
- They can be applied to page sidebars for showing individual actions, handling local navigation, and displaying metadata
```

File: content/components/action-list.mdx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Action lists support section dividers and headers for grouping items, and indivudal item dividers for added clarity.
+
+Action lists use a mobile-friendly inset style. Their sizes are adapted on touch devices, and their single-column format should render consistently in any screen size.
```
Comment (yaili) [inline content/components/action-list.mdx:26]: Could it be:
Their size adapts to touch devices.

Or:
Their size adjusts to touch devices.

Comment (yaili) [reply content/components/action-list.mdx:26]:
```suggestion
Action lists use a mobile-friendly inset style. Their sizes are adapted on touch devices, and their single-column format renders consistently in any screen size.
```

File: content/components/action-list.mdx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Except for read-only and disabled exceptions, action list items are interactive.
```
Comment (yaili) [inline content/components/action-list.mdx:28]:
```suggestion
Action list items are interactive, with the exception of read-only and disabled items.
```

File: content/components/action-list.mdx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+### In overlays
+
+### In pages
+
+
+-->
+
+## Anatomy 
+
+<img src="https://user-images.githubusercontent.com/293280/122128726-d7692b80-cde9-11eb-8e5b-2f4e2eb4b4a1.png" alt="Action list anatomy" />
```
Comment (yaili) [inline content/components/action-list.mdx:56]: I think we need a small caption here, even if just:

An action list is composed of:
- Action list item
- Item dividers
- Section divider (subtle or filled)
- Section header (subtle or filled)

File: content/components/action-list.mdx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        <Flex justifyContent="center">
+            <img width="464" alt="Action list sizes" src="https://user-images.githubusercontent.com/293280/121766871-c1b5e680-cb09-11eb-8835-bfec4c80e45a.png" />
+        </Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Sizes</Heading>
+        <Flex as="p">Action list items support three different sizes: small, medium, and large. The small size is the default and most common option. Medium sizes work well for relaxed local navigation, while large sizes can support items that need more breathing room.</Flex>
+        <Flex as="p">Sizes only grow vertically. This behavior keeps the content aligned among items, and retains horizontal space for density.</Flex>
+        <Flex as="p">On touch devices, the large size is used at all times to ensure usability when tapping.</Flex>
+        <Flex as="p">Avoid mixing different sizes in the same list.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:74]: "Avoid mixing" or "Do not mix"?

File: content/components/action-list.mdx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+            </Box>
+            <Box>
+                <img width="464" alt=" " src="https://user-images.githubusercontent.com/293280/121966449-8c84e080-cd23-11eb-930f-9397ae2ea433.png" />
+                <Caption>Use leading visuals to represent content types.</Caption>
+            </Box>
+        </Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Leading visual</Heading>
+        <Flex as="p">Leading visuals are optional and appear at the start of an item. They can be octicons, avatars, and other custom visuals that fit a small area.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:94]: Would it look odd if the leading visuals section was text on the left and images on the right?

File: content/components/action-list.mdx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        <Flex as="p">When listing system sections, features, or options, use leading visuals to improve the items' scannability. Or, in user-provided objects, they can help signifying the item's content type and status.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:95]:
```suggestion
        <Flex as="p">When listing system sections, features, or options, use leading visuals to improve the items' scannability. In user-generated objects, they can help indicate the item's content type and status.</Flex>
```

File: content/components/action-list.mdx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        <Flex as="p">Depending on the context however, displaying a leading visual may not be necessary. Per instance, a list of branches in a select panel may not need repeated icons if the surrounding UI already gives enough hints about its content type.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:96]:
```suggestion
        <Flex as="p">Depending on the context, displaying a leading visual may not be necessary. For example, a list of branches in a select panel may not need repeated icons if the surrounding UI provides enough hints about its content type.</Flex>
```

File: content/components/action-list.mdx [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    </div>
+    <div>
+        <Flex justifyContent="center">&nbsp;</Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Trailing visual and trailing text</Heading>
+        <Box as="p">Trailing visual and trailing text can display auxiliary information. They're placed at the right of the item, and can denote status, keyboard shortcuts, or be used to set expectations about what the action does.</Box>
+        <Box as="p">Use an `arrow-right` octicon in menus to indicate the action will open more options, such as in a nested context. Or, use an `edit` octicon to denote the item is gonna be edited after clicking it.</Box>
```
Comment (yaili) [inline content/components/action-list.mdx:104]:
```suggestion
        <Box as="p">Use an `arrow-right` octicon in menus to indicate the action will open more options, such as in a nested context. Use an `edit` octicon to indicate the item is going to be edited after clicking it.</Box>
```

File: content/components/action-list.mdx [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+        <Box as="p">Custom trailing elements are supported, such as counters, labels, and other custom visuals that may help identifying the item.</Box>
```
Comment (yaili) [inline content/components/action-list.mdx:105]:
```suggestion
        <Box as="p">Custom trailing elements are supported, such as counters, labels, and other custom visuals that may help identify the item.</Box>
```

File: content/components/action-list.mdx [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+        <Box as="p">When using a trailing text for displaying keyboard shortcuts, always confirm the characters match with the user's OS. For example, for denoting a bold action in a Markdown toolbar, Use "Ctrl+B" on Linux and Windows, and "⌘B" on Mac. (<a href="https://apple.stackexchange.com/questions/55727/where-can-i-find-the-unicode-symbols-for-mac-functional-keys-command-shift-e">See reference for Mac functional keys</a>).</Box>
```
Comment (yaili) [inline content/components/action-list.mdx:106]:
```suggestion
        <Box as="p">When using a trailing text for displaying keyboard shortcuts, always confirm the characters match with the user's operating system. For example, to indicate a bold action in a Markdown toolbar, use "Ctrl+B" on Linux and Windows, and "⌘B" on Mac. (<a href="https://apple.stackexchange.com/questions/55727/where-can-i-find-the-unicode-symbols-for-mac-functional-keys-command-shift-e">See reference for Mac functional keys</a>).</Box>
```

File: content/components/action-list.mdx [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+    </div>
+    <div>
+        <Flex justifyContent="center">&nbsp;</Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Danger</Heading>
+        <Flex as="p">An action list item can have a special "danger" style, to be used in special cases that are destructive and require double attention from the user.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:113]:
```suggestion
        <Flex as="p">An action list item can have a special "danger" style, to be used in special cases that are destructive and require extra attention from the user.</Flex>
```

File: content/components/action-list.mdx [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+        <Flex as="p">Place danger items at the end of the list.</Flex>
+    </div>
+    <div>
+        <Flex justifyContent="center">&nbsp;</Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Item dividers</Heading>
+        <Flex as="p">Item dividers allow users to parse heavier amounts of information. They're placed between items and are useful in complex lists, particularly when descriptions or multi-line text is present.</Flex>
+        <Flex as="p">When considering whether to use item dividers, make sure they actually make the presented information easier to parse, and not only add noise to the UI.</Flex>
```
Comment (yaili) [inline content/components/action-list.mdx:122]:
```suggestion
        <Flex as="p">When considering whether to use item dividers, make sure they truly make the presented information easier to parse, instead of adding visual clutter to the interface.</Flex>
```

**Review by yaili — COMMENTED** [review]
Some copy edits suggestions @vdepizzol 💖

File: content/components/action-list.mdx (hunk 5 above)
Comment (vdepizzol) [reply content/components/action-list.mdx:76]: I think it's a message on the "caution" side instead of fully prohibitive. Do you think that'd work in this case?

File: content/components/action-list.mdx [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+### In overlays
+
+### In pages
+
+
+-->
+
+## Anatomy
+
+An action list can be composed with:
```
Comment (yaili) [inline content/components/action-list.mdx:69]:
```suggestion
An action list can be composed of:
```

File: content/components/action-list.mdx [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+        <Box as="p">Dropdown menus are used for making a single selection among a small list of options. The list is usually predefined with system values, but in some cases it can include user-provided data when those are known not to grow into too many items.</Box>
+    </div>
+    <div>
+        <Link href="/components/select-panel">
+            <img alt="Select panel" src="https://user-images.githubusercontent.com/293280/123876997-4158f900-d8f1-11eb-85cb-461d9bbee0cb.png" />
+        </Link>
+    </div>
+    <div>
+        <Link href="/components/select-panel" fontWeight="bold">Select panel</Link>
+        <Box as="p">Select panels allows manipulating long lists of options, with filtering and other advanced interactions. They can be used for single or multi-selection.</Box>
```
Comment (yaili) [inline content/components/action-list.mdx:193]:
```suggestion
        <Box as="p">Select panels allow manipulating long lists of options, with filtering and other advanced interactions. They can be used for single or multi-selection.</Box>
```

File: content/components/action-list.mdx (hunk 14 above)
Comment (yaili) [inline content/components/action-list.mdx:69]:
```suggestion
An action list can be composed of:
```

File: content/components/index.mdx [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+<Grid gridTemplateColumns={['1fr', null, null, null, '1fr 2fr']} gridGap={4}>
+    <div>
+        <Link href="/components/action-list">
+            <img alt="Select panel" src="https://user-images.githubusercontent.com/293280/123880841-54bb9280-d8f8-11eb-9a6f-4c33c8196fbf.png" />
+        </Link>
+    </div>
+    <div>
+        <Link href="/components/action-list" fontWeight="bold">Action list</Link>
+        <Box as="p">An action list is a vertical list of interactive actions or options. It’s composed of items presented in a consistent single-column format, with room for icons, descriptions, side information, and other rich visuals.</Box>
+    </div>
```
Comment (yaili) [inline content/components/index.mdx:17]: From here down, we need to either comment out or move to another PR @vdepizzol

File: content/components/action-list.mdx [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+
+<img src="https://user-images.githubusercontent.com/293280/122133975-96c1e000-cdf2-11eb-9ce9-02d5014c1831.png" alt="Action list examples" />
+
+## Overview
+
+Action lists can have many applications:
+- They’re the foundation of menus, selection panels, and other overlay components
+- They can be applied to page sidebars for showing individual actions, handling local navigation, and displaying metadata
+
+Action lists support section dividers and headers for grouping items, and indivudal item dividers for added clarity.
```
Comment (yaili) [inline content/components/action-list.mdx:37]:
```suggestion
Action lists support section dividers and headers for grouping items, and individual item dividers for added clarity.
```

File: content/components/action-list.mdx [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+                <img width="464" alt=" " src="https://user-images.githubusercontent.com/293280/124064900-9298e380-d9ea-11eb-8aa9-dae40ece9d5b.png" />
+                <Caption>Trailing text with custom styling to indicate diff change.</Caption>
+            </Box>
+        </Flex>
+    </div>
+    <div>
+        <Heading fontSize={3}>Trailing visual and trailing text</Heading>
+        <Box as="p">Trailing visual and trailing text can display auxiliary information. They're placed at the right of the item, and can denote status, keyboard shortcuts, or be used to set expectations about what the action does.</Box>
+        <Box as="p">Note these side visuals don't have dedicated interaction targets.</Box>
+        <Box as="p">Use an `arrow-right` octicon in menus to indicate the action will open more options, such as in a nested context. Use an `edit` octicon to indicate the item is going to be edited after clicking it.</Box>
```
Comment (yaili) [inline content/components/action-list.mdx:134]: We need figure out why these code blocks aren't rendering in the preview and either fix or use normal text.

File: content/components/action-list.mdx [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+        <Box as="p">An action list item can have a special "danger" style, to be used in cases that require extra attention from the user.</Box>
+        <Box as="p">For destructive or irremediable actions, show a confirmation dialog for extra friction. If the action is not destructive, present the user a way to undo the action instead of asking for confirmation. <a href="https://alistapart.com/article/neveruseawarning/">Never use a warning when you mean undo</a>.</Box>
+        <Box as="p">Place danger items at the end of the list.</Box>
+    </div>
+</Grid>
+
+## Application
+
+### In overlays
+
```
Comment (yaili) [inline content/components/action-list.mdx:166]: The components listed below can't have a link until the page exists. Can we comment this section out, or remove the links before we publish? @vdepizzol

File: content/components/action-list.mdx (hunk 18 above)
Comment (colebemis) [reply content/components/action-list.mdx:134]: In MDX, text is not parsed as markdown unless it's surrounded by newlines. So backticks won't be treated as inline code unless the containing text block is surrounded by newlines. Give this a try:

```suggestion
        <Box as="p">
       
Use an `arrow-right` octicon in menus to indicate the action will open more options, such as in a nested context. Use an `edit` octicon to indicate the item is going to be edited after clicking it.

        </Box>
```

Comment (vdepizzol) [reply content/components/action-list.mdx:134]: @colebemis It works, thanks! 🥳

File: content/components/action-list.mdx (hunk 19 above)
Comment (vdepizzol) [reply content/components/action-list.mdx:199]: Fixed!

File: content/components/index.mdx (hunk 16 above)
Comment (vdepizzol) [reply content/components/index.mdx:17]: Done!

**Comment by vdepizzol** [comment]
@yaili I've cleaned up this PR and removed the content originally referring to pages that are not yet available! ✨ 

While there's more we could write about action lists (e.g. accessibility and application in sidebars), I think the content we have now is in great shape to go live iteratively.

While we don't have links to interface guidelines for action menus, dropdown menus and select panels, I've included links to the React implementation for now, I think it's better than not talking about it. Let me know if that works, we can adjust it as we work on these other guideline pages:

> Screen Shot 2021-07-16 at 15 09 44

💖 💖

**Review by simurai — APPROVED** [review]
